scripts/README.md

```diff
@@ -0,0 +1,174 @@
+# KubeCF support scripts
+
+## Overview
+
+The `scripts/` directory contains scripts to support KubeCF development, setup
+kubernetes via `kind` or `minikube`, and build, deploy, and test KubeCF.
+
+All scripts expect to be invoked from the root of the `kubecf` git repository.
+It contains a `Makefile` that can be used to launch them (tab-completion).
+
+Many scripts will only work from a git checkout because the product versioning
+needs to look up tags.
+
+## Shared Configuration
+
+The script use a common framework via `source scripts/include/setup.sh`. It
+provides some shared configuration settings:
+
+### `XTRACE`
+
+Setting `XTRACE=1 make ...` will turn on bash tracing for the scripts. This
+is mostly a debugging tool when the scripts fail, and the cause isn't obvious.
+
+### `PINNED_TOOLS`
+
+Each script declares all external tools it is using with a command like this:
+
+```bash
+require_tools cf kubectl
+```
+
+Minimum tool versions, and potential download sources are defined in `./scripts/tools`.
+It contains a [README](tools/README.md) that explains the various ways to
+define them.
+
+By default the scripts will accept the minimum or any later version that
+can be invoked via `PATH` and will only download and install a newer version
+if the local version is missing or too old.
+
+If `PINNED_TOOLS` is set to a non-empty value, then an exact version match is
+required.
+
+### `TOOLS_DIR`
+
+Downloaded tools will be installed into `TOOLS_DIR`, which defaults to
+`./output/bin`. This directory will always be at the front of the `PATH`
+when these scripts are executed.
+
+## Index of scripts
+
+
+### Development
+
+#### ./scripts/tools-install.sh
+
+This script installs the pinned version of all configured tools into `TOOLS_DIR`
+unless the installed version is already an exact match.
+
+#### ./scripts/tools-versions.sh
+
+List all configured tools and their current versions.
```

Comment:
Maybe worth documenting how this behaves when PINNED_TOOLS is set. E.g. If I have pinned tools and I have a version of a tool installed locally (but not the exact required version), what will this script return, my installed version or missing or other?